via/static/scripts/video_player/components/Transcript.tsx

```diff
@@ -385,22 +396,30 @@ export default function Transcript({
               'bg-grey-3/30'
             )}
           >
-            {transcript.segments.map((segment, index) => (
-              <TranscriptSegment
-                key={index}
-                hidden={
-                  filterMatches
-                    ? !filterMatches.has(index) && index !== currentIndex
-                    : false
-                }
-                highlight={highlight}
-                isCurrent={index === currentIndex}
-                matches={filterMatches?.get(index)}
-                onSelect={() => onSelectSegment?.(segment)}
-                time={segment.start}
-                text={segment.text}
-              />
-            ))}
+            {transcript.segments.map((segment, index, segments) => {
+              const nextSegment =
+                index < segments.length - 1 ? segments[index + 1] : undefined;
```

Comment:
I think `const nextSegment = segments[index + 1]` should have the same result, as `segments[index + 1]` will be undefined when `index >= segments.length`.

```suggestion
              const nextSegment = segments[index + 1];
```

Reply:
It does at runtime, but not in TS's type system, at least with our current TS config. Doing an explicit check allows TS to know that `nextSegment` might be undefined.